tests/std/tests/P0896R4_ranges_alg_swap_ranges/test.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+constexpr void smoke_test() {
+    using ranges::swap_ranges, ranges::swap_ranges_result, ranges::iterator_t;
+    using std::same_as;
+    using I1 = iterator_t<move_only_range<int>>;
+    using I2 = iterator_t<move_only_range<int>>;
+
+    // Validate that swap_ranges_result aliases in_in_result
+    STATIC_ASSERT(same_as<ranges::swap_ranges_result<int, double>, ranges::in_in_result<int, double>>);

```

Comment:
Should this take advantage of the `using`-declaration for `ranges::swap_ranges_result` above?